### PR TITLE
Support the openSUSE AArch64 JeOS flavor

### PR DESCRIPTION
**Pull request by Vogtinator**
Using an unfortunately complex test suite setup, a test converts the image, and the chained tests boot from it and install jeos-firstboot for proper JeOS testing.

It already found a product bug!

Verification runs:
http://10.161.32.77/tests/31
http://10.161.32.77/tests/32

### CODE REVIEW

**Comment by Vogtinator** [comment]
@thehejik: If this approach with a new var (set by https://gitlab.suse.de/openqa/scripts/merge_requests/299) is merged, it would also require changes for the SLE test triggering.

**Comment by thehejik** [comment]
LGTM - but we would need also to modify rsync.pl for SLE to send new needed variable  HDD_1_NAME

File: lib/main_common.pm [hunk 1]
```diff
@@ -544,6 +544,11 @@ sub load_jeos_tests {
         return;
     }
     unless (get_var('LTP_COMMAND_FILE')) {
+        if (check_var('ARCH', 'aarch64') && is_opensuse()) {
+            # Enable jeos-firstboot, due to boo#1020019
+            load_boot_tests();
+            loadtest "jeos/prepare_firstboot";
```
Comment (okurz) [inline lib/main_common.pm]: And you want to call `load_boot_tests` still afterwards?

File: tests/jeos/prepare_firstboot.pm [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    assert_script_run("touch /var/lib/YaST2/reconfig_system");
+    assert_script_run("systemctl enable jeos-firstboot");
+
+    # Remove current root password
+    assert_script_run("sed -i 's/^root:[^:]*:/root:*:/' /etc/shadow", 600);
+
+    # Reboot
+    type_string("reboot\n");
+
+    return;
```
Comment (okurz) [inline tests/jeos/prepare_firstboot.pm]: I don't think you need this

File: tests/jeos/prepare_firstboot.pm [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    zypper_call('in jeos-firstboot');
+
+    # Enable jeos-firstboot for next boot
+    assert_script_run("touch /var/lib/YaST2/reconfig_system");
+    assert_script_run("systemctl enable jeos-firstboot");
+
+    # Remove current root password
+    assert_script_run("sed -i 's/^root:[^:]*:/root:*:/' /etc/shadow", 600);
+
+    # Reboot
```
Comment (okurz) [inline tests/jeos/prepare_firstboot.pm]: seems redundant. Mind to delete?

File: tests/jeos/prepare_firstboot.pm [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+use utils 'zypper_call';
+
+sub run {
+    # Login with default credentials (root:linux)
+    assert_screen('linux-login', 300);
+    type_string("root\n",  wait_still_screen => 5);
+    type_string("linux\n", wait_still_screen => 5);
+
+    # Ensure YaST2-Firstboot is disabled, as we use jeos-firstboot in openQA
+    script_run("systemctl disable YaST2-Firstboot");
```
Comment (okurz) [inline tests/jeos/prepare_firstboot.pm]: why not use `assert_script_run`?

File: tests/jeos/prepare_firstboot.pm [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+    # Ensure jeos-firstboot is installed
```
Comment (okurz) [inline tests/jeos/prepare_firstboot.pm]: comment seems stating the obvious, mind to delete?

File: lib/main_common.pm (hunk 1 above)
Comment (Vogtinator) [reply lib/main_common.pm]: Yup, the test triggers a reboot

**Pull request merged**